Fixed #3645 - User defined smtp settings not used when sending email

dawansv opened the pull request:
Added inbound_email_id when sending mail in order to get correct smtp settings

## Description
fix #3645 

## Motivation and Context
Changes needed to send from user specified smtp settings

## How To Test This
1. You need a way to know which SMTP server is used when sending. An easy way is to use gmail smtp because it will always set the from_address to the smtp user; make sure you use one gmail or google apps account for the system smtp 
2. Under a test user, create a new entry under Outgoing SMTP Mail Servers under Profile - scroll all the way down - Settings - Mail Accounts; here make sure to use another gmail or google apps account so that it will be clear whether that smtp user is used or not when receiving email.
2. Create or edit the corresponding incoming account under Mail account, select SMTP settings created previously. Save all and close.
3. Compose new email and send to an email address you can access. Check email and confirm it is from user smtp userid, not from system smtp userid.

## Types of changes
- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

daniel-samson commented on modules/Emails/vardefs.php:
```diff
@@ -483,6 +483,16 @@
             'inline_edit' => false,
         ),
 
+        'inbound_email_id' => array(
```
This field is not needed. The mailbox_id field already stores the id of the inbound email record.

daniel-samson commented on modules/Emails/Email.php:
```diff
@@ -2697,7 +2697,8 @@ public function send()
         }
 
-        $mail = $this->setMailer($mail);
+        $ieId = $this->inbound_email_id;
```
use mailbox_id instead.

Dillon-Brown commented:
Works as expected

pgorod commented:
@Dillon-Brown you know you can make your assessments as code reviews instead of simple comments?

You can find that button `Review changes`  on the `Files changed` tab of the PR. Even unprivileged users like me can do it.

Unless this part of your flow to do it like this...

dawansv commented:
@daniel-samson 

Thanks. Missed that one. Will update.

On a side not, I wanted to test my changes with the latest hotfix so I first did a resync of my local hotfix branch with the upstream branch. But then I get an "underfined" error when sending an email or trying to update smtp settings in the admin interface.  Went back to my origin fork from a few days ago and works fine. To be clear I am just talking hotfix branches here without any of my changes.

I am doing something stupid or are other experiencing same problem on current hotfix branch? Easier way to test is just to try to edit smtp settings from the admin - Outbound Email option.

Here is the compare between my original hotfix branch and the curent one:
https://github.com/dawansv/SuiteCRM/compare/hotfix...salesagility:hotfix

Will have to test more and submit a new bug if I can confirm this.

dawansv commented:
@daniel-samson 

Never mind the undefined error. It's a problem on my side. Sorry about the noise.

dawansv commented:
Add to modify controller.php to set mailbox_id in action_send() because it is not populated automatically since it's different from the REQUEST variable name.

Small conflict in controller.php because of some end of line difference, but I can't figure out how to get rid of it... Grrrr.

daniel-samson commented on modules/Emails/EmailsController.php:
```diff
@@ -135,6 +135,7 @@ public function action_ComposeView()
     public function action_send()
     {
         $this->bean = $this->bean->populateBeanFromRequest($this->bean, $_REQUEST);
+        $this->bean->mailbox_id = $_REQUEST['inbound_email_id'];
```
Can you please move this into the populateBeanFromRequest? So that we can reuse this code as much as possible.

I have just noticed that this method isn't using the $request variable. We will need to update this.

dawansv commented:
@daniel-samson 

OK I moved the mailbox_id assignment to populateBeanFromRequest.  I also removed it from function actionSaveDraft() in EmailController.php since it is now superfluous there as well. I did a grep -r * -e "mailbox_id" on the whole tree and as far as I can tell, there is no similar assignment anywhere else.

As for the assignment in populateBeanFromRequest, I used $_REQUEST instead of $request because the rest of the code does the same.  It looks like this function is pending cleanup as relating to the use of $request (as per comment in header), but I am not sure which direction you want it to go...

samus-aran commented:
@dawansv From what I can review it looks fine the execution.  Are you complete and we can get this re-tested internally?
